sources/engine/Stride.UI/Controls/ScrollingText.cs

```diff
@@ -190,6 +195,10 @@ private void UpdateAndAdjustDisplayText(GameTime time = null)
                     nextLetterIndex = 0;
 
                 textToDisplayWidth += addedCharacterWidth;
+
+                // workaround to avoid endless loop
+                if (addedCharacterWidth <= 0.0f)
+                    break;
```

Comment:
So, from what I understand, this component displays an 'infinitely' repeating text horizontally. The issue happens when the whole text has a width of less than or equal to zero. It fills the component with characters until it reaches the max width but since the text's width is effectively zero it can't ever finish that loop.

If I did understand that correctly, a cleaner solution would be to just take the width of the whole text that would be included and if that width is below an epsilon avoid the whole loop/drawing the text, this can be done through ``CalculateTextSize(YourText).X``.
As, right now, from what I understood of your workaround, it seems to bail out as soon as *just one* of the characters has a zero width ? I think there are a couple of languages where zero-width characters are useful, not sure how applicable it is to our engine but might as well account for it ( [Zero-width_joiner](https://en.wikipedia.org/wiki/Zero-width_joiner) and [Zero-width_non-joiner](https://en.wikipedia.org/wiki/Zero-width_non-joiner) perhaps ? )